File: PackScript.Tests/Integration/SyncTests.cs
using System.IO;
using FluentAssertions;
using NUnit.Framework;
using PackScript.Core.Infrastructure;
using PackScript.Tests.TestInfrastructure;

namespace PackScript.Tests.Integration
{
    [TestFixture]
    public class SyncTests
    {
        private TestFilesApi api;
        private PackContext context;

        [TestFixtureSetUp]
        public void Setup()
        {
            api = new TestFilesApi();
            context = ContextFactory.Create(@"..\..\Integration\Sync", api).ScanForResources().BuildAll();
        }

        [Test]
        public void Simple()
        {
            api.copyFileCalls[0].FirstArg.Should().Be(FullPath("test.js"));
            api.copyFileCalls[0].SecondArg.Should().Be(FullPath("Target/Simple/test.js"));
        }

        [Test]
        public void Child()
        {
            api.copyFileCalls[1].FirstArg.Should().Be(FullPath(@"Child\test.js"));
            api.copyFileCalls[1].SecondArg.Should().Be(FullPath("Target/Child/test.js"));
        }

        [Test]
        public void Recursive()
        {
            api.copyFileCalls[2].FirstArg.Should().Be(FullPath(@"test.js"));
            api.copyFileCalls[2].SecondArg.Should().Be(FullPath("Target/Recursive/test.js"));
            api.copyFileCalls[3].FirstArg.Should().Be(FullPath(@"Child\test.js"));
            api.copyFileCalls[3].SecondArg.Should().Be(FullPath("Target/Recursive/Child/test.js"));
        }

        [Test]
        public void Alternate()
        {
            api.copyFileCalls[4].FirstArg.Should().Be(FullPath(@"test.js"));
            api.copyFileCalls[4].SecondArg.Should().Be(FullPath("Target/Alternate/test.js"));
        }

        private string FullPath(string path)
        {
            return Path.GetFullPath(@"..\..\Integration\Sync\") + path;
        }
    }
}

File: PackScript.Node/Tests/Integration/Sync.tests.js
integrationTest('Sync', function (output) {
    var copyFile = Pack.api.Files.copyFile;
    equal(copyFile.firstCall.args[0], 'Tests/Integration/Sync/test.js');
    equal(copyFile.firstCall.args[0], fullPath('Sync/test.js'), 'simple');
    equal(copyFile.firstCall.args[1], fullPath('TestOutput/Sync/Simple/test.js'), 'simple');

    equal(copyFile.secondCall.args[0], fullPath('Sync/Child/test.js'), 'child');
    equal(copyFile.secondCall.args[1], fullPath('TestOutput/Sync/Child/test.js'), 'child');

    equal(copyFile.thirdCall.args[0], fullPath('Sync/test.js'), 'recursive');
    equal(copyFile.thirdCall.args[1], fullPath('TestOutput/Sync/Recursive/test.js'), 'recursive');
    equal(copyFile.getCall(3).args[0], fullPath('Sync/Child/test.js'), 'recursive');
    equal(copyFile.getCall(3).args[1], fullPath('TestOutput/Sync/Recursive/Child/test.js'), 'recursive');

    equal(copyFile.getCall(4).args[0], fullPath('Sync/test.js'), 'alternate');
    equal(copyFile.getCall(4).args[1], fullPath('TestOutput/Sync/Alternate/test.js'), 'alternate');

    function fullPath(path) {
        return 'Tests/Integration/' + path;
    }
});
